### PR TITLE
Update reproducible comparing tests with centos:7

### LINKED ISSUE #3687: Update reproducible comparing tests on linux (not docker image)
Part of https://github.com/adoptium/temurin-build/issues/3683

- [x] SCM_REF build issue https://github.com/adoptium/temurin-build/issues/3710
- [x] sbom with personal build branch and sha
- [x] Retrieving boot JDK 21.0.2+13-LTS not available -- BOOTJDK version is not abvailable by API https://ci.adoptium.net/view/Test_grinder/job/Grinder/9300/ -- remove suffix -LTS if bootjdk include -LTS
- [x] error: pathspec '"e95140bf52915bda6c5ff271538789ed9a55162b\ne95140bf52915bda6c5ff271538789ed9a55162b\ne95140bf52915bda6c5ff271538789ed9a55162b\ne95140bf52915bda6c5ff271538789ed9a55162b\ne95140bf52915bda6c5ff271538789ed9a55162b"' did not match any file(s) known to git. https://github.com/adoptium/temurin-build/issues/3687#issuecomment-2018898703
- [ ] Support running with upsteam jenkins jobs by mount test resources hit the user permission issue https://github.com/adoptium/infrastructure/issues/3490

### PR DESCRIPTION
**Pull request by sophia-guo**
Update reproducible comparing tests with centos:7

Fix #3687

### CODE REVIEW

**Comment by sophia-guo** [comment]
https://ci.adoptium.net/view/Test_grinder/job/Grinder/9326/

**Comment by karianna** [comment]
@sophia-guo linter failures to start with.

**Comment by smlambert** [comment]
The code changes look good to me, though I am wondering why there are these differences that cause the test to fail:

```
23:30:14  Comparing ...
23:30:14  diff -r jdk-22+36/release compare.1/jdk-22+36/release
23:30:14  11,13d10
23:30:14  < BUILD_SOURCE="git:e95140bf52915bda6c5ff271538789ed9a55162b"
23:30:14  < BUILD_SOURCE_REPO="https://github.com/adoptium/temurin-build.git"
23:30:14  < SOURCE_REPO="https://github.com/adoptium/jdk22.git"
23:30:14  Differences found..., logged in: reprotest.diff
```

My query does not stop us from merging this PR.

**Review by smlambert — APPROVED** [review]

**Comment by andrew-m-leonard** [comment]
> The code changes look good to me, though I am wondering why there are these differences that cause the test to fail:
> 
> ```
> 23:30:14  Comparing ...
> 23:30:14  diff -r jdk-22+36/release compare.1/jdk-22+36/release
> 23:30:14  11,13d10
> 23:30:14  < BUILD_SOURCE="git:e95140bf52915bda6c5ff271538789ed9a55162b"
> 23:30:14  < BUILD_SOURCE_REPO="https://github.com/adoptium/temurin-build.git"
> 23:30:14  < SOURCE_REPO="https://github.com/adoptium/jdk22.git"
> 23:30:14  Differences found..., logged in: reprotest.diff
> ```
> 
> My query does not stop us from merging this PR.

@sophia-guo I think since this is a reproducible build comparison, the version of temurin-build repo being used should not differ, it would imply the 2nd build is built with a different version of the build scripts...? Do you know why that may be the case?

File: tooling/reproducible/linux_repro_build_compare.sh [hunk 1]
```diff
@@ -61,9 +62,13 @@ setEnvironment() {
 }
 
 cleanBuildInfo() {
+  # shellcheck disable=SC3043
+  local DIR="$1"
   # BUILD_INFO name of OS level build was built on will likely differ
-  sed -i '/^BUILD_INFO=.*$/d' "jdk-${TEMURIN_VERSION}/release"
-  sed -i '/^BUILD_INFO=.*$/d' "compare.$$/jdk-${TEMURIN_VERSION}/release"
+  sed -i '/^BUILD_SOURCE=.*$/d' "${DIR}/release"
```
Comment (andrew-m-leonard) [inline tooling/reproducible/linux_repro_build_compare.sh:68]: I'm now thinking this is not right, we should be using the identical BUILD_SOURCE, ie the exact same temurin-build scripts version...?

File: test/system/reproducibleCompare/build.xml [hunk 2]
```diff
@@ -14,32 +14,68 @@
...
+		<if>
+			<isset property="env.VENDOR_TEST_BRANCHES"/>
+			<then>
+				<property name="buildBranch" value="${env.VENDOR_TEST_BRANCHES}"/>
+			</then>
+			<else>
+				<property name="buildBranch" value="master"/>
+			</else>
+		</if>
+		<exec executable="git" failonerror="true">
```
Comment (andrew-m-leonard) [inline test/system/reproducibleCompare/build.xml:58]: is this cloning and checking out the exact correct version of temurin-build as the original JDK was built with?

**Review by andrew-m-leonard — CHANGES_REQUESTED** [review]

**Comment by sophia-guo** [comment]
> https://github.com/adoptium/temurin-build/pull/3739#issuecomment-2038423300

I'm not sure if those differences are expected. Was confirmed that those should be cleaned. https://github.com/adoptium/temurin-build/issues/3687#issuecomment-2034828779

@andrew-m-leonard so we shouldn't clean those information? To be clear the reproduced JDK release file doesn't have any of those build repo information at all. 

[release.txt](https://github.com/adoptium/temurin-build/files/14886153/release.txt)

**Comment by andrew-m-leonard** [comment]
> > [#3739 (comment)](https://github.com/adoptium/temurin-build/pull/3739#issuecomment-2038423300)
> 
> I'm not sure if those differences are expected. Was confirmed that those should be cleaned. [#3687 (comment)](https://github.com/adoptium/temurin-build/issues/3687#issuecomment-2034828779)
> 
> @andrew-m-leonard so we shouldn't clean those information? To be clear the reproduced JDK release file doesn't have any of those build repo information at all.
> 
> [release.txt](https://github.com/adoptium/temurin-build/files/14886153/release.txt)

Hmm, that would imply it's built differently than the original

File: tooling/reproducible/linux_repro_build_compare.sh (hunk 1 above)
Comment (andrew-m-leonard) [reply tooling/reproducible/linux_repro_build_compare.sh:68]: right, I see now, so yes this is being build by linux_repro_build_compare.sh, not the build_farm which adds BUILD_SOURCE , BUILD_REPO, BUILD_SOURCE_REPO

File: test/system/reproducibleCompare/build.xml (hunk 2 above)
Comment (andrew-m-leonard) [reply test/system/reproducibleCompare/build.xml:58]: done by linux_repro_build.sh

Comment (sophia-guo) [reply test/system/reproducibleCompare/build.xml:58]: Reproduced JDK is using the parameters from SBOM file, which has nothing to do with this build step. This step ensures that files under temurin-build/test/system/reproducibleCompare and temurin-build/tooling/reproducible/ are using the specified repo and branch. Though if the specified repo and branch does change the file linux_repo_build_compare.sh the reproducing and  comparing process or rule may change. However it won't change the parameters from SBOM file.

**Review by andrew-m-leonard — APPROVED** [review]
think I am good now, lets get this in and try it

**Comment by sophia-guo** [comment]
If build is unstable that means there are difference. The reproduced jdk tar file is also archived. https://ci.adoptium.net/view/Test_grinder/job/Grinder/9364/

File: tooling/reproducible/linux_repro_build_compare.sh (hunk 1 above)
Comment (sophia-guo) [reply tooling/reproducible/linux_repro_build_compare.sh:68]: Just to confirm so should I remove those three lines or keep as is?

Comment (sophia-guo) [reply tooling/reproducible/linux_repro_build_compare.sh:68]: @andrew-m-leonard

**Pull request merged**